### DISCUSSION TITLE
How can I handle errors with RLS permissions?

### DISCUSSION THREAD

**Discussion by alstep08** [supabase/supabase#8431] [Questions] [answered]
I am using supabase-js client with react app. I want to grant access to delete or update data to a limited category of users by creating row-level security rules. The rules work fine, but I can't handle errors on the client.

If a user without delete rights tries to delete an entry, then nothing happens, a response with a status of 200 and without any errors is returned, but the deletion does not occur. The same happens when updating rows.

**Comment by GaryAustin1** [comment, marked as answer]
This is the way Postgres RLS works with anything involving select (and update needs select permission).  If you do a select you will get no clue if the row exists, or RLS blocked it.  The same applies to Update.

If you Google something like "postgres row level security update no error" you will find some discussions on it.  One suggestion is that you can put a trigger function on the update and basically check your RLS condition in it and generate an error, if that is your desire.

**Comment by steve-chavez** [comment]
> If a user without delete rights tries to delete an entry

RLS permissions work as Gary mentioned. However if the user doesn't have the right to delete any entry at all, using regular table-level security(GRANT/REVOKE) would result in an error.
